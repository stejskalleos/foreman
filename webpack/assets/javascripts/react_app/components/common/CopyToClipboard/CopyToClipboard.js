import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import { OverlayTrigger, Tooltip, Button } from 'patternfly-react';
import { CopyIcon } from '@patternfly/react-icons';
import { translate as __ } from '../../../common/I18n';

const CopyToClipboard = ({ valueToCopy }) => {
  const inputRef = useRef(null);
  const tooltip = <Tooltip id="copy-btn-tooltip">{__('Copied!')}</Tooltip>;

  const handleCopy = () => {
    inputRef.current.select();
    document.execCommand('copy');
  };

  return (
    <div className="input-group">
      <input
        className="form-control"
        id="input_to_copy"
        readOnly
        type="text"
        value={valueToCopy}
        ref={inputRef}
      />
      <div className="input-group-btn">
        <OverlayTrigger
          placement="top"
          overlay={tooltip}
          trigger="click"
          rootClose
        >
          <Button bsStyle="default" onClick={handleCopy} className="pull-right">
            <CopyIcon /> {__('Copy')}
          </Button>
        </OverlayTrigger>
      </div>
    </div>
  );
};

CopyToClipboard.propTypes = {
  valueToCopy: PropTypes.string,
};

CopyToClipboard.defaultProps = {
  valueToCopy: null,
};

export default CopyToClipboard;
