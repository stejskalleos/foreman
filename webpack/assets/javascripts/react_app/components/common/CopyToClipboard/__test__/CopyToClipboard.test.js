import { testComponentSnapshotsWithFixtures } from '../../../../common/testHelpers';

import CopyToClipboard from '../CopyToClipboard';

const fixtures = {
  ok: {
    valueToCopy: 'Some value to copy!',
  },
};

describe('CopyToClipboard', () => {
  testComponentSnapshotsWithFixtures(CopyToClipboard, fixtures);
});
