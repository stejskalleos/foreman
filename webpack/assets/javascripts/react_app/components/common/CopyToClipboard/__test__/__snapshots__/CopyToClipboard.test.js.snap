// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyToClipboard ok 1`] = `
<div
  className="input-group"
>
  <input
    className="form-control"
    id="input_to_copy"
    readOnly={true}
    type="text"
    value="Some value to copy!"
  />
  <div
    className="input-group-btn"
  >
    <OverlayTrigger
      defaultOverlayShown={false}
      overlay={
        <Tooltip
          bsClass="tooltip"
          id="copy-btn-tooltip"
          placement="right"
        >
          Copied!
        </Tooltip>
      }
      placement="top"
      rootClose={true}
      trigger="click"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="pull-right"
        disabled={false}
        onClick={[Function]}
      >
        <CopyIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
         
        Copy
      </Button>
    </OverlayTrigger>
  </div>
</div>
`;
